mapreduce/src/test/java/uk/gov/gchq/palisade/PalisadeInputSplitTest.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2018 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.gchq.palisade;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import uk.gov.gchq.palisade.resource.Resource;
+import uk.gov.gchq.palisade.resource.StubResource;
+import uk.gov.gchq.palisade.service.Service;
+import uk.gov.gchq.palisade.service.request.ConnectionDetail;
+import uk.gov.gchq.palisade.service.request.DataRequestResponse;
+import uk.gov.gchq.palisade.service.request.StubConnectionDetail;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class PalisadeInputSplitTest {
+
+    @Test(expected = IOException.class)
+    public void shouldntAcceptNegativeLength() throws IOException {
+        //Given
+        ByteArrayOutputStream buffer = new ByteArrayOutputStream();
+        DataOutputStream dos = new DataOutputStream(buffer);
+        dos.writeInt(-1);
+        //When
```

Comment:
Not a big issue, but please could you move these lines up to the 'given' section

```java
ByteArrayInputStream bis = new ByteArrayInputStream(buffer.toByteArray());
DataInputStream dis = new DataInputStream(bis);
PalisadeInputSplit test = new PalisadeInputSplit();
```

That way the 'when' section just contains the single line that we are testing and at a glance it is easy to tell what method we are invoking.

Reply:
Done